src/function.ts

```diff
@@ -61,28 +125,539 @@ export class Function<P, O> implements Integration {
+  ) {
+    if (!this.scope) {
+      throw Error("Must first be bound to a Construct using .bind().");
+    }
+    const scope = this.scope;
+    let tokens: string[] = [];
+    const preWarmContext = new NativePreWarmContext(this.props);
+    const func = this.func(preWarmContext);
+
+    const result = await serializeFunction(
+      // factory function allows us to prewarm the clients and other context.
+      () => {
+        integrationPrewarms.forEach((i) => i?.(preWarmContext));
+        return func;
+      },
+      {
+        isFactoryFunction: true,
+        serialize: (obj) => {
+          if (typeof obj === "string") {
+            const reversed =
+              Tokenization.reverse(obj, { failConcat: false }) ??
+              Tokenization.reverseString(obj).tokens;
+            if (!Array.isArray(reversed) || reversed.length > 0) {
+              if (Array.isArray(reversed)) {
+                tokens = [...tokens, ...reversed.map((s) => s.toString())];
+              } else {
+                tokens = [...tokens, reversed.toString()];
+              }
+            }
+          } else if (typeof obj === "object") {
+            /**
+             * Remove unnecessary fields from {@link CfnResource} that bloat or fail the closure serialization.
+             */
+            const transformCfnResource = (o: unknown): any => {
+              if (Resource.isResource(o as any)) {
+                const { node, stack, env, ...rest } = o as unknown as Resource;
+                return rest;
+              } else if (CfnResource.isCfnResource(o as any)) {
+                const {
+                  stack,
+                  node,
+                  creationStack,
+                  // don't need to serialize at runtime
+                  _toCloudFormation,
+                  // @ts-ignore - private - adds the tag manager, which we don't need
+                  cfnProperties,
+                  ...rest
+                } = transformTable(o as CfnResource);
+                return transformTaggableResource(rest);
+              } else if (Token.isUnresolved(o)) {
+                const token = (<any>o).toString();
+                // add to tokens to be turned into env variables.
+                tokens = [...tokens, token];
+                return token;
+              }
+              return o;
+            };
+
+            /**
+             * Remove unnecessary fields from {@link CfnTable} that bloat or fail the closure serialization.
+             */
+            const transformTable = (o: CfnResource): CfnResource => {
+              if (
+                o.cfnResourceType ===
+                aws_dynamodb.CfnTable.CFN_RESOURCE_TYPE_NAME
+              ) {
+                const table = o as aws_dynamodb.CfnTable;
+                if (!table.streamSpecification) {
+                  const { attrStreamArn, ...rest } = table;
+
+                  return rest as unknown as CfnResource;
+                }
```

Comment:
What's the logic here? Seems like we're keeping a whole bunch

Reply:
the table is also going through the CfnResource filtering transform, this just adds the safety for the streamArm.

Reply:
Updated the docs.